File: shared/src/commonMain/kotlin/com/browsy/data/repository/LocalBookShelfStorage.kt
package com.browsy.data.repository

/**
 * Platform-specific storage for book shelf data.
 *
 * This expect class defines the contract for persistent storage of saved books.
 * Platform implementations use appropriate storage mechanisms:
 * - Android: SharedPreferences
 * - iOS: NSUserDefaults
 */
internal expect class LocalBookShelfStorage() {
    /**
     * Saves the JSON string representation of saved books.
     * @param data JSON string containing list of SavedBook objects
     */
    fun save(data: String)

    /**
     * Loads the previously saved JSON string.
     * @return JSON string if data exists, null otherwise
     */
    fun load(): String?
}

File: iosApp/iosApp/Data/Repository/LocalBookShelfStorage.swift
import Foundation

/// Platform-specific storage for book shelf data using UserDefaults.
class LocalBookShelfStorage {
    private let userDefaults = UserDefaults.standard
    private let storageKey = "saved_books_data"
    
    /// Saves the JSON string representation of saved books.
    func save(_ data: String) {
        userDefaults.set(data, forKey: storageKey)
    }
    
    /// Loads the previously saved JSON string.
    func load() -> String? {
        return userDefaults.string(forKey: storageKey)
    }
}
